docs/internals/annotations_resolving.md

```diff
@@ -0,0 +1,267 @@
+!!! note
+    This section is part of the *internals* documentation, and is partly targeted to contributors.
+
+Pydantic heavily relies on [type hints][type hint] at runtime to build schemas for validation, serialization, etc.
+
+While type hints were primarily introduced for static type checkers (such as [Mypy] or [Pyright]), they are
+accessible (and sometimes evaluated) at runtime. This means that the following would fail at runtime,
+because `Node` has yet to be defined in the current module:
+
+```python test="skip" lint="skip"
+class Node:
+    """Binary tree node."""
+
+    # NameError: name 'Node' is not defined:
+    def __init__(self, l: Node, r: Node) -> None:
+        self.left = l
+        self.right = r
+```
+
+To circumvent this issue, forward references can be used (by wrapping the annotation into quotes).
+
+In Python 3.7, [PEP 563] introduced the concept of _postponed evaluation of annotations_, meaning
+with the `from __future__ import annotations` [future statement], type hints are stringified by default:
+
+```python requires="3.12" lint="skip"
+from __future__ import annotations
+
+from pydantic import BaseModel
+
+
+class Foo(BaseModel):
+    f: MyType
+    # Without the future import, equivalent to:
+    # f: 'MyType'
```

Comment:
Maybe this would be more clear as like "given the future import above, this is equivalent to f: 'MyType'" or something like that?